src/main/java/com/amazon/opendistroforelasticsearch/security/OpenDistroSecurityPlugin.java

```diff
@@ -881,8 +881,12 @@ public Settings additionalSettings() {
             settings.add(Setting.listSetting(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_IGNORE_REQUESTS, Collections.emptyList(), Function.identity(), Property.NodeScope)); //not filtered here
             settings.add(Setting.boolSetting(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_RESOLVE_BULK_REQUESTS, false, Property.NodeScope, Property.Filtered));
             settings.add(Setting.boolSetting(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_EXCLUDE_SENSITIVE_HEADERS, true, Property.NodeScope, Property.Filtered));
-    
-            
+            settings.add(Setting.listSetting(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_CONFIG_READONLY_FIELDS, Collections.emptyList(), Function.identity(), (list) -> {
+                if (!AuditConfig.FIELD_PATHS.containsAll(list)) {
+                    throw new IllegalArgumentException("Invalid resource paths for read-only configs passed.");
+                }
```

Comment:
Should we fail or ignore with a message in case the path does not exist ?

In the future if you remove one of the existing paths, it might cause ES to not start anymore if this setting was not updated.

Reply:
I think it is better to fail.
The fields are updated to dynamically fetch all valid fields now. If customer has configured something wrong, then customers may ignore message and wonder why it is not working correctly. 

If some path is completely removed in the future, then it will force user to migrate which is desirable IMO. 
Any thoughts ?